benchmark/fs/bench-readSync.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+tmpdir.refresh();
+
+const tmpfile = tmpdir.resolve(`.existing-file-${process.pid}`);
+fs.writeFileSync(tmpfile, 'this-is-for-a-benchmark', 'utf8');
+
+const bench = common.createBenchmark(main, {
+  type: ['existing', 'non-existing'],
+  paramType: ['offset-and-length', 'no-offset-and-length'],
+  n: [1e4],
+});
+
+function main({ n, type, paramType }) {
+  let fd;
+
+  switch (type) {
+    case 'existing':
+      fd = fs.openSync(tmpfile, 'r', 0o666);
+      break;
+    case 'non-existing':
+      fd = 1 << 30;
+      break;
+    default:
+      new Error('Invalid type');
+  }
+
+  bench.start();
+  switch (paramType) {
+    case 'offset-and-length':
+      for (let i = 0; i < n; i++) {
+        try {
+          fs.readSync(fd, Buffer.alloc(1), 0, 1, 0);
```

Comment:
I think the buffer should be reused, as what's normally done in the wild. Also "reading a file from offset 1 many many times, one character at a time" do not seem to be a very typical use case. It would be better to tweak the benchmark so that it creates a big file in a reused buffer (with a length of `fs.statSync(file).blksize`) and then read without offset until the file ends. We probably do not need an additional case for the offset as reading a file repeatedly from the same offset is probably not common enough in the wild to optimize for.

Reply:
Thank you for your good comments.
I tried modifying it, but I'm not sure if it worked as you said.
Would it be better to remove `paramType` and test it by passing only buffer?